By yanniboi: Added GDPR consent module for tracking user consent.

yanniboi opened the pull request.

Commit:
By yanniboi: Added GDPR consent module for tracking user consent to data storage.

Commit:
By yanniboi: Updated composer dependencies.

Commit:
By yanniboi: PHPCS review.

Commit:
By yanniboi: PHPCS review.

mhavelant commented on modules/gdpr_consent/src/Controller/ConsentAgreementController.php:
```diff
@@ -0,0 +1,266 @@
+   *   An array suitable for drupal_render().
+   */
+  public function revisionShow($gdpr_consent_agreement_revision) {
+    $gdpr_consent_agreement = $this->entityManager()
```
entityManager() is deprecated, we shouldn't use it.

mhavelant commented on modules/gdpr_consent/src/Controller/ConsentAgreementController.php:
```diff
@@ -0,0 +1,266 @@
+      ->loadRevision($gdpr_consent_agreement_revision);
+    return $this->t('Revision of %title from %date', [
+      '%title' => $gdpr_consent_agreement->label(),
+      '%date' => format_date($gdpr_consent_agreement->getRevisionCreationTime()),
```
format_date is deprecated, we shouldn't use it.

mhavelant commented:
Adding a consent field to a content type, then trying to add content results in an exception.
`
The website encountered an unexpected error. Please try again later.</br></br><em class="placeholder">Exception</em>: Could not determine user ID for entity type node. Please ensure there is a resolver registered. in <em class="placeholder">Drupal\gdpr_consent\ConsentUserResolver\ConsentUserResolverPluginManager-&gt;getForEntityType()</em> (line <em class="placeholder">95</em> of <em class="placeholder">modules/custom/gdpr/modules/gdpr_consent/src/ConsentUserResolver/ConsentUserResolverPluginManager.php</em>). <pre class="backtrace">Drupal\gdpr_consent\Plugin\Field\FieldWidget\ConsentWidget-&gt;getConsentingUser(Object) (Line: 41)
Drupal\gdpr_consent\Plugin\Field\FieldWidget\ConsentWidget-&gt;formElement(Object, 0, Array, Array, Object) (Line: 335)
Drupal\Core\Field\WidgetBase-&gt;formSingleElement(Object, 0, Array, Array, Object) (Line: 202)
...
`

Commit:
By yanniboi: Fixed service injection in Consent controller.

Commit:
By yanniboi: Added node entity consent resolver.

yanniboi commented:
I've fixed the exception. We use consent user resolver plugins to determine how to get the relevant user from the entity that the field is on. We did not yet have a resolver for node entities, so I added one in my most recent commit https://github.com/brainsum/gdpr/pull/6/commits/937fac7904f0cf98496a2b4ec28f2374c73d2ad6

Commit:
By yanniboi: PHPCS review.

mhavelant commented:
I tested it, looks good!

The only issue I found is that creating an explicit agreement implies that the user has to check it, but it's not a required checkbox. But this can be added later.
